lib/galaxy/webapps/galaxy/buildapp.py

```diff
@@ -72,6 +72,7 @@ def app_factory(global_conf, load_app_kwds={}, **kwargs):
     webapp.add_route('/authnz/{provider}/login', controller='authnz', action='login', provider=None)
     webapp.add_route('/authnz/{provider}/callback', controller='authnz', action='callback', provider=None)
     webapp.add_route('/authnz/{provider}/disconnect', controller='authnz', action='disconnect', provider=None)
+    webapp.add_route('/authnz/logout', controller='authnz', action='logout')
```

Comment:
Should this be a per provider logout as well?

Reply:
I think having per provider logout could work too, but this seemed more natural to me from the perspective of the UI. For login, the user has picked a specific provider to log into. However, on logout the user has only indicated that they want to logout. /authnz/logout figures out what provider was used to authenticate (using the `identity-provider` cookie) and logs the user out of that provider.

Reply:
The UI perspective makes sense. What do you think about the API perspective? I feel as if having a per provider logout allows programmatic logout from specific providers, although I'm not sure there's a clear use case. Perhaps a programmatic global logout from a specific provider like keycloak only through the API perhaps?  Is the only necessary change to read the cookie from the client side instead of the server?  I also feel like `authnz/logout` should log the user out from all providers, not just the last logged in one? Just some thoughts, but no strong views on which may be better.

Reply:
What your saying makes a lot of sense. I do like that per provider logout is consistent with the other provider methods.

> Is the only necessary change to read the cookie from the client side instead of the server?

Yeah I think that's basically it. However, I think I would change it so that server side the cookie is used to determine the logout API URL and this is provided to the client. I'll work on updating the PR.

Reply:
Hi @nuwang , I looked into passing a value to the UI, but it seemed simpler and more in keeping with the Galaxy client design to add an API method for getting the provider specific logout url, so I did that. The frontend now calls `/authnz/logout` which reads the oidc-provider cookie and if it exists, redirects to `/authnz/{provider}/logout`.